com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -152,21 +149,13 @@ public static string[] GetOutputNames(Model model)
                 return failedModelChecks;
             }
 
-            foreach (var constantName in TensorNames.RequiredConstants)
+            var modelApiVersionTensor = model.GetTensorByName(TensorNames.VersionNumber);
+            if (modelApiVersionTensor == null)
```

Comment:
Again, doesn't need to be in this PR, but I feel like we lost some of the checks that the tensors we need are in the Model. It's never an issue for model files that we create, but a frequent source of forum/github posts would be people trying to train their own models, then add them for inference and get null reference errors because the assumed the tensors were there without checking.

I need to look a little more closely, but I think things like HasContinuousOutputs() would throw if not of the expected tensors are there.

We could add something like

```
bool CheckExpectedTensors(List<string> errorMessagesOut) {}
```
to the extension methods, call it once here, and then assume everything is present after that.

We should also have a test that CheckModel() with a basically empty model doesn't throw and returns some error strings.

Feel free to just log a jira as a reminder to come back to this, since I think it's a small regression in usability.

Reply:
I think I took care of that by the order of the checks, but I agree we should check it explicitly for safety. 
Made a [JIRA ticket](https://jira.unity3d.com/browse/MLA-1565).